inject.go

```diff
@@ -69,33 +85,44 @@ func Inject(target interface{}) Option {
 	// 		target.Field(2),  // Baz io.Writer
 	// 	]
 	//
-	// As we iterate through the arguments received by the function, we can
-	// simply copy the value into the corresponding value in the targets list.
+	// As we iterate through the fields of the generated struct, we can simply
+	// copy the value into the corresponding value in the targets list.
 	targets := make([]reflect.Value, 0, t.NumField())
 
 	for i := 0; i < t.NumField(); i++ {
 		f := t.Field(i)
-		// Skip private fields.
-		if f.PkgPath != "" {
+
+		// Skip private fields
+		if f.PkgPath != "" && !f.Anonymous {
```

Comment:
This will panic:
```golang
package main

import (
	"fmt"

	"go.uber.org/fx"
)

type x struct {
	field string
}

type y struct {
	x
}

func main() {
	var Y y
	err := fx.New(
		fx.Provide(func() x {
			return x{"alex"}
		}),
		fx.Inject(&Y),
	)
	fmt.Println(err)
} 
```
Here is the output:
```
tmp ❯ go run main.go
2017/07/25 19:06:20 [Fx] PROVIDE	main.x <= main.main.func1()
2017/07/25 19:06:20 [Fx] PROVIDE	fx.Lifecycle <= go.uber.org/fx.New.func1()
2017/07/25 19:06:20 [Fx] INVOKE		reflect.makeFuncStub()
panic: reflect: reflect.Value.Set using value obtained using unexported field

goroutine 1 [running]:
reflect.flag.mustBeAssignable(0x1d9)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:225 +0x190
reflect.Value.Set(0x10ec140, 0xc42000e2d0, 0x1d9, 0x10ec140, 0xc42000a830, 0x99)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:1339 +0x2f
go.uber.org/fx.Inject.func2(0xc42000a800, 0x1, 0x1, 0xc42000a800, 0x0, 0x1)
	/Users/alexsamylkin/gocode/src/go.uber.org/fx/inject.go:139 +0x10b
reflect.callReflect(0xc42000a540, 0xc42003d868)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:508 +0x38f
reflect.makeFuncStub(0x0, 0x0, 0x110769e, 0x4, 0xc42003dbd0, 0x108dc3f, 0xc420016510, 0xc42000a540, 0xc42000a7e0, 0x2000000020, ...)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/asm_amd64.s:17 +0x40
reflect.Value.call(0xc42006c0c0, 0xc42000a540, 0x13, 0x1107636, 0x4, 0xc42000a6e0, 0x1, 0x1, 0x0, 0xc42006c0c0, ...)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:434 +0x91f
reflect.Value.Call(0xc42006c0c0, 0xc42000a540, 0x13, 0xc42000a6e0, 0x1, 0x1, 0x0, 0x0, 0x80)
	/usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:302 +0xa4
go.uber.org/fx/vendor/go.uber.org/dig.(*Container).Invoke(0xc42000e3a0, 0xc42006c0c0, 0xc42000a540, 0x0, 0x0, 0x0, 0x1, 0x1)
	/Users/alexsamylkin/gocode/src/go.uber.org/fx/vendor/go.uber.org/dig/dig.go:128 +0x31d
go.uber.org/fx.(*App).executeInvokes(0xc42006c120, 0x10dda40, 0xc42000e480)
	/Users/alexsamylkin/gocode/src/go.uber.org/fx/app.go:239 +0x3f3
go.uber.org/fx.New(0xc42003df58, 0x2, 0x2, 0xc42000a560)
	/Users/alexsamylkin/gocode/src/go.uber.org/fx/app.go:207 +0x40a
main.main()
	/Users/alexsamylkin/gocode/src/tmp/main.go:24 +0x150
```